src/main/java/com/ibm/disni/rdma/verbs/impl/RdmaVerbsNat.java

```diff
@@ -142,20 +128,14 @@ public IbvQP modifyQP(IbvQP qp, IbvQpAttr attr)
 		return null;
 	}
 
-	public SVCRegMr regMr(IbvPd pd, ByteBuffer buffer, int access) throws IOException {
-		NatRegMrCall regMrCall = regList.poll();
-		if (regMrCall == null){
-			regMrCall = new NatRegMrCall(this, nativeDispatcher, memAlloc);
-		}
+	public SVCRegMr regMr(IbvPd pd, ByteBuffer buffer, int access) {
+		NatRegMrCall regMrCall =  new NatRegMrCall(this, nativeDispatcher, memAlloc);
 		regMrCall.set(pd, buffer, access);
 		return regMrCall;
 	}
 
-	public SVCRegMr regMr(IbvPd pd, long address, int length, int access) throws IOException {
-		NatRegMrCall regMrCall = regList.poll();
-		if (regMrCall == null){
-			regMrCall = new NatRegMrCall(this, nativeDispatcher, memAlloc);
-		}
+	public SVCRegMr regMr(IbvPd pd, long address, int length, int access) {
+		NatRegMrCall regMrCall = new NatRegMrCall(this, nativeDispatcher, memAlloc);
 		regMrCall.set(pd, address, length, access);
 		return regMrCall;
```

Comment:
Just noticed that regMr still implements the old model where the SVC object is created and then the parameters are passed via "set" method which is public. Why don't we make all SVC calls consistent and pass the params via constructor, just like we did for postSend postRecv, etc. The same inconsistency also occurs in some other methods (deregMr, reqNotifyCQ).